adapters/YahooAdapter/YahooAdapter/Rewarded/GADMAdapterYahooRewardedAd.m

```diff
@@ -0,0 +1,194 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterYahooRewardedAd.h"
+#import <YahooAds/YahooAds.h>
+#include <stdatomic.h>
+#import "GADMAdapterYahooConstants.h"
+#import "GADMAdapterYahooUtils.h"
+
+NSString *const GADMAdapterVerizonVideoCompleteEventId = @"onVideoComplete";
+
+@interface GADMAdapterYahooRewardedAd () <YASInterstitialAdDelegate>
+@end
+
+@implementation GADMAdapterYahooRewardedAd {
+  /// The completion handler to call when the ad loading succeeds or fails.
+  GADMediationRewardedLoadCompletionHandler _adLoadCompletionHandler;
+
+  /// Yahoo rewarded ad.
+  YASInterstitialAd *_rewardedAd;
+
+  /// Placement ID string used to request ads from Yahoo Mobile SDK.
+  NSString *_placementID;
+
+  /// An ad event delegate to invoke when ad rendering events occur.
+  __weak id<GADMediationRewardedAdEventDelegate> _adEventDelegate;
+
+  /// Ad Configuration for the ad to be rendered.
+  GADMediationRewardedAdConfiguration *_adConfiguration;
+
+  BOOL _isVideoCompletionEventCalled;
+}
+
+- (void)loadRewardedAdForAdConfiguration:(nonnull GADMediationRewardedAdConfiguration *)adConfig
```

Comment:
I verified RewardedAd with one of our test ads, the video plays and all callbacks are working correctly.